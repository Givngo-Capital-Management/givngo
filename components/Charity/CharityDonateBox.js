import React from 'react';
import { Button, Grid } from 'semantic-ui-react';
import styles from './CharityDonateBox.module.scss';

export default function CharityDonateBox() {
  return (
    <div className={styles.box}>
      <div className={styles.header}>
        <p>Donate now</p>
      </div>
      <div className={styles.content}>
        <Grid>
          <div className="equal width row">
            <div className="ten wide column">
              <h3 className="ui header">
                <div className="content">
                  Donate to Charity name
                  <div className="sub header">Charity short description</div>
                </div>
              </h3>
            </div>
            <div className="six wide column">
              <Button className="donate ui right floated" color="orange">
                Donate
              </Button>
            </div>
          </div>
        </Grid>

        <h3>Total raised: $132,619</h3>
      </div>
    </div>
  );
}
