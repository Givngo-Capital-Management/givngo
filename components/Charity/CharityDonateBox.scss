.charity-donate-box {
  display: block;
  border: 1px solid #ccc;
  margin-bottom: 24px; 
}

.charity-donate-box-header {
  padding: 16px;
  line-height: 2rem;
  background-color: rgb(241, 241, 241);
  font-size: 1.2rem;
  font-weight: 400;
}

.charity-donate-box-content {
  border-top: 1px solid #ccc;
  padding: 16px;
}