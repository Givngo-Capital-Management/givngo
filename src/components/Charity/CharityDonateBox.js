import React from 'react';
import { Button, Grid } from 'semantic-ui-react';
import DonateModal from '../Stripe/DonateModal';
import styles from './CharityDonateBox.module.scss';

export default function CharityDonateBox(props) {
  const { charity } = props

  const donateButton = (
    <Button className="donate ui right floated" color="orange">
      Donate
    </Button>
  )

  return (
    <div className={styles.box}>
      <div className={styles.header}>
        <p>Donate now</p>
      </div>
      <div className={styles.content}>
        <Grid>
          <div className="equal width row">
            <div className="ten wide column">
              <h3 className="ui header">
                <div className="content">
                  {`Donate to ${charity.name}`}
                </div>
              </h3>
            </div>
            <div className="six wide column">
              <DonateModal charity={charity} button={donateButton}/>
            </div>
          </div>
        </Grid>

        <h3>{`Total raised: $${charity.total_raised || 0}`}</h3>
      </div>
    </div>
  );
}
