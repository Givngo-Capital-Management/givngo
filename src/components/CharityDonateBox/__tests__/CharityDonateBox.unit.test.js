import React from 'react';
import { shallow } from 'enzyme';
import CharityDonateBox from '../CharityDonateBox';

describe('CharityDonateBox', () => {
  test('renders', () => {
    const wrapper = shallow(<CharityDonateBox />);
    expect(wrapper).toMatchSnapshot();
  });
});
