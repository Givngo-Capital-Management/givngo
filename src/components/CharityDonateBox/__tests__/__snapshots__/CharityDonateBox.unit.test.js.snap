// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharityDonateBox renders 1`] = `
<div
  className="charity-donate-box"
>
  <div
    className="charity-donate-box-header"
  >
    <p>
      Donate now
    </p>
  </div>
  <div
    className="charity-donate-box-content"
  >
    <div
      className="ui equal width grid"
    >
      <div
        className="equal width row"
      >
        <div
          className="ten wide column"
        >
          <h3
            className="ui header"
          >
            <div
              className="content"
            >
              Donate to Charity name
              <div
                className="sub header"
              >
                Charity short description
              </div>
            </div>
          </h3>
        </div>
        <div
          className="six wide column"
        >
          <Button
            as="button"
            className="donate ui right floated"
            color="orange"
          >
            Donate
          </Button>
        </div>
      </div>
    </div>
    <h3>
      Total raised: $132,619
    </h3>
  </div>
</div>
`;
